rdflib/graph.py

```diff
@@ -1510,10 +1509,17 @@ def do_de_skolemize(uriref, t):
 
         def do_de_skolemize2(t):
             (s, p, o) = t
-            if isinstance(s, Genid):
-                s = s.de_skolemize()
-            if isinstance(o, Genid):
-                o = o.de_skolemize()
+
+            if Genid._is_external_skolem(s):
+                s = Genid(s).de_skolemize()
+            elif RDFLibGenid._is_rdflib_skolem(s):
+                s = RDFLibGenid(s).de_skolemize()
```

Comment:
I'm going to add tests to confirm, but I think the second branch here `elif RDFLibGenid._is_rdflib_skolem(s)` may never execute, as the condition for `Genid._is_external_skolem(s)` will always be true if `RDFLibGenid._is_rdflib_skolem(s)` is true.

Reply:
Actually I may only get to this tomorrow, there are quite a backlog of issues I'm trying to fix right now, but I before we have a test specifically verifying this I would not be comfortable with merging.

Reply:
Feel free to add it in the meantime, I will let you know when I start working on this.

Reply:
I think you are right. We should do the `RDFLibGenid._is_rdflib_skolem` check before `Genid._is_external_skolem`. I'll add a test to confirm.